src/libsyntax/parse/parser.rs

```diff
@@ -5394,13 +5397,19 @@ impl<'a> Parser<'a> {
                           &format!("maybe move this module `{0}` to its own directory \
                                      via `{0}/mod.rs`",
                                     this_module));
-            if paths.path_exists {
+            if legacy_warn {
+                err.note(
+                    "this was erroneously allowed and will become a hard error in a future release"
```

Comment:
Hmm. I think we could do an actual lint here -- can't we leave a little note on the AST or something and then issue the lint later, in the lint pass?

At minimum, though, can you open a "forward compat" issue with the appropriate labels and structure and mention it in this message? Roughly follow [the protocol described in RFC 1589](https://github.com/rust-lang/rfcs/blob/master/text/1589-rustc-bug-fix-procedure.md)